src/contributors/MetricsTableContributor.ts

```diff
@@ -313,7 +328,63 @@ export class MetricsTableContributor extends Contributor {
                 });
             }
         }
-        this.selectedTerms = Array.from(termsSet);
+
+        /** This block verifies if selected terms exist in data, fetches the data if so.
+         * Then it adds a row to metricsTable in order to have a complete table.
+         */
+        if (termsSet.size > 0) {
+            const missingRows = [];
+            const dataRows = new Set(...this.computableResponse?.metricsResponse.map(r => r.keys));
+            termsSet.forEach(term => {
+                if (!dataRows.has(term)) {
+                    missingRows.push(term);
+                }
+            });
+            if (missingRows.length > 0) {
+                this.isDataUpdating = true;
+                return forkJoin(this.table.vectors.map(v =>
```

Comment:
why do you need to return something as you subscribe into the method ? Its to unsubscribe  ?

Reply:
you are right, fixing